envi/tests/test_arch_arm.py

```diff
@@ -5,58 +5,594 @@
 import envi.registers as e_reg
 import envi.memcanvas as e_memcanvas
 import envi.memcanvas.renderers as e_rend
-import envi.archs.h8 as e_h8
+import envi.archs.arm as arm
 import vivisect
 import platform
-import unittest
+import unittest  
 from envi import IF_RET, IF_NOFALL, IF_BRANCH, IF_CALL, IF_COND
-from envi.archs.h8.regs import *
-from envi.archs.h8.const import *
-from envi.archs.h8.parsers import *
+from envi.archs.arm.regs import *
+from envi.archs.arm.const import *
+from envi.archs.arm.disasm import *
+import binascii   # temporarily included for testing purposes only - allows to print out binary values
 
 
-# OPHEX, VA, repr, flags, emutests
-instrs = [
-        ('08309fe5', 0xbfb00000, 'ldr r3, [#0xbfb00010]', 0, ()),
-        ('0830bbe5', 0xbfb00000, 'ldr r3, [r11, #0x8]!', 0, ()),
-        ('08309be4', 0xbfb00000, 'ldr r3, [r11], #0x8', 0, ()),
-        ('08301be4', 0xbfb00000, 'ldr r3, [r11], #-0x8', 0, ()),
-        ('02209ae7', 0xbfb00000, 'ldr r2, [r10, r2]', 0, ()),
-        ('02209ae6', 0xbfb00000, 'ldr r2, [r10], r2', 0, ()),
-        ('02203ae7', 0xbfb00000, 'ldr r2, [r10, -r2]!', 0, ()),
-        ('0220bae7', 0xbfb00000, 'ldr r2, [r10, r2]!', 0, ()),
-        ('22209ae7', 0xbfb00000, 'ldr r2, [r10, r2 lsr #32]', 0, ()),
-        ('08309fe5', 0xbfb00000, 'ldr r3, [#0xbfb00010]', 0, ()),
-        ('08309fe5', 0xbfb00000, 'ldr r3, [#0xbfb00010]', 0, ()),
-        ]
-# FIXME: create list of this from IDA flow below - THIS CURRENT DATA IS FOR H8!  NOT ARM/THUMB
+
+
+''' 
+
+  This dictionary will contain all commands supported by ARM to test
+  Fields will contain following information:
+  archBitMask, ophex, va, repr, flags, emutests
 '''
-        ( '8342', 0x4560, 'add.b #42, r3h', IF_B, () ),
```

Comment:
Only keeping this section for reference. I will cut it down further. I just need the setup and test versions for examples. Will be removed as soon as first emulation test is included.

Date: Fri, 3 Jun 2016 05:55:51 -0700
From: notifications@github.com
To: vivisect@noreply.github.com
CC: redheadedrod@hotmail.com; author@noreply.github.com
Subject: Re: [atlas0fd00m/vivisect] Update unit tests for opcodes (#3)

In envi/tests/test_arch_arm.py:

>  '''
> -        ( '8342', 0x4560, 'add.b #42, r3h', IF_B, () ),

these, however, are obviously H8 remnants and can be deleted.


—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.